Sources/Core/UI/Viewers/AV/AVViewController.swift

```diff
@@ -0,0 +1,153 @@
+    var toolbarButtons: [UIBarButtonItem] = []
+    
+    private var url: URL
+    private var file: File?
+    private var AVPlayerVC: AVPlayerViewController
+    private var client: BoxClient?
+    
+    private lazy var titleView: CustomTitleView = {
+        let view: CustomTitleView = CustomTitleView()
+        return view
+    }()
+    
+    // MARK: - Initializer
+    
+    public init(url: URL, file: File? = nil, client: BoxClient? = nil, actions: [FileInteractions]) {
+        self.url = url
+        self.file = file
+        self.AVPlayerVC = AVPlayerViewController()
+        self.client = client
+        super.init(nibName: nil, bundle: nil)
+        // Not showing controls, so when video players doesn't flash a play button before it automatically starts playing
+        self.AVPlayerVC.showsPlaybackControls = false
+        self.setupPlayer()
+    }
+    
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: - View life cycle
+    
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+    }
+}
+
+// MARK: - Setup video player
+
+private extension AVViewController {
+    func setupPlayer() {
+        if let unwrappedClient = client {
+            self.getToken(client: unwrappedClient) { result in
+                switch result {
+                case let .success(accessToken):
+                    let headers: [String: String] = [
+                       "Authorization": "Bearer \(accessToken)"
+                    ]
+                    DispatchQueue.main.async {
+                        let asset = AVURLAsset(url: self.url, options: ["AVURLAssetHTTPHeaderFieldsKey": headers])
+                        let playerItem = AVPlayerItem(asset: asset)
+                        self.AVPlayerVC.player = AVPlayer(playerItem: playerItem)
+                        self.AVPlayerVC.player?.play()
+                        self.AVPlayerVC.showsPlaybackControls = true
+                    }
+                case .failure:
+                    DispatchQueue.main.async {
+                        self.showAlertWith(title: "Error", message: "Unable to connect to Box account.")
+                    }
+                }
+            }
+        }
+        else {
+            let asset = AVURLAsset(url: url)
+            let playerItem = AVPlayerItem(asset: asset)
+            DispatchQueue.main.async {
+                self.AVPlayerVC.player = AVPlayer(playerItem: playerItem)
+                self.AVPlayerVC.player?.play()
+                self.AVPlayerVC.showsPlaybackControls = true
+            }
+        }
+    }
+    
+    // Gets a fresh token that will last for around an hour if the session allows for a new token
+    // swiftlint:disable cyclomatic_complexity
+    func getToken(client: BoxClient, completion: @escaping (Result<String, BoxSDKError>) -> Void) {
```

Comment:
Move to previewHelper class?  Then we can pull out ref to client in this View Controller.